utils/build-script

```diff
@@ -740,6 +780,23 @@ the number of parallel build jobs to use""",
                 "--symbols-package.")
             return 1
 
+    if args.android:
+        if args.android_ndk is None or \
+                args.android_ndk_version is None or \
+                args.android_icu_uc is None or \
+                args.android_icu_uc_include is None or \
+                args.android_icu_i18n is None or \
+                args.android_icu_i18n_include is None:
+            print_with_argv0("When building for Android, --android-ndk, "
+                             "--android-ndk-version, --android-icu-uc, "
+                             "--android-icu-uc-include, --android-icu-i18n, "
+                             "and --android-icu-i18n-include must be "
+                             "specified.")
+            return 1
+        # lib/Driver/ToolChains.cpp depends on this environment variable being
```

Comment:
> The variable is consumed within the CMAKE invocations

I don't think so.  We pass the CMake variable separately on its command line.

> We could also pass these paths in individually from build-swift or elsewhere

I would very much prefer that, assuming that's what Clang does.  Having the compiler behavior depend on environment variables seems very fragile to me, and I can see how it will lead to hard-to-debug situations for our users.  Say, you see a line in the build log.  You copy it into the terminal, and it does not work.

> this seems to be the cleanest option

"Fewer changes to existing systems" is not always the cleanest and maintainable option.